### PR TITLE
Make all enabled ADCs available in the shell

### PR DESCRIPTION
**Pull request by benediktibk**
Previously, only a single type of ADCs was always available in the ADC shell. This change enables the usage of the ADC shell for different ADC types at the same time.

### CODE REVIEW

File: drivers/adc/adc_shell.c [hunk 1]
```diff
@@ -89,32 +48,84 @@ LOG_MODULE_REGISTER(adc_shell);
...
-/* This table size is = ADC devices count + 1 (NA). */
 static struct adc_hdl {
 	const struct device *dev;
 	struct adc_channel_cfg channel_config;
 	uint8_t resolution;
 } adc_list[] = {
-	FOR_EACH(ADC_HDL_LIST_ENTRY, (,), INIT_MACRO())
+#define DT_DRV_COMPAT atmel_sam_afec
+	DT_INST_FOREACH_STATUS_OKAY(ADC_HDL_LIST_ENTRY)
+#undef DT_DRV_COMPAT
```
Comment (anangl) [inline drivers/adc/adc_shell.c:74]: Wouldn't this:
```suggestion
	DT_FOREACH_STATUS_OKAY(atmel_sam_afec, ADC_HDL_LIST_ENTRY)
```
work with `ADC_HDL_LIST_ENTRY` defined as:
```
#define ADC_HDL_LIST_ENTRY(node_id)					\
	{								\
		.dev = DEVICE_DT_GET(node_id),				\
		...
```
?

Comment (benediktibk) [reply drivers/adc/adc_shell.c]: Way better, done.

File: drivers/adc/adc_shell.c [hunk 2]
```diff
@@ -89,32 +48,48 @@ LOG_MODULE_REGISTER(adc_shell);
...
-#define DEVICES(n) DEVICE_DT_INST_GET(n),
-#define ADC_HDL_LIST_ENTRY(dev_)					\
-	{								\
-		.dev = dev_,						\
-		.channel_config = {					\
-			.gain = ADC_GAIN_1,				\
-			.reference = ADC_REF_INTERNAL,			\
-			.acquisition_time = ADC_ACQ_TIME_DEFAULT,	\
-			.channel_id = 0,				\
-		},							\
```
Comment (anangl) [inline drivers/adc/adc_shell.c:101]: What was wrong with this?

**Review by anangl — APPROVED** [review]

File: drivers/adc/adc_shell.c (hunk 2 above)
Comment (benediktibk) [reply drivers/adc/adc_shell.c:101]: I touched the macro define and let clang-format run over it to fix the indentation of the line breaks. I intentionally skipped eveything else clang-format changed in this file and staged only formatting changes where I changed something by myself.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**